js/audio/shared/DeviceSelect.svelte

```diff
@@ -0,0 +1,69 @@
+<script lang="ts">
+	import RecordPlugin from "wavesurfer.js/dist/plugins/record.js";
+	import type { I18nFormatter } from "@gradio/utils";
+	import { createEventDispatcher } from "svelte";
+
+	export let i18n: I18nFormatter;
+	export let micDevices: MediaDeviceInfo[] = [];
+
+	const dispatch = createEventDispatcher<{
+		error: string;
+	}>();
+
+	$: try {
+		let tempDevices: MediaDeviceInfo[] = [];
+		RecordPlugin.getAvailableAudioDevices().then(
+			(devices: MediaDeviceInfo[]) => {
+				micDevices = devices;
+				devices.forEach((device) => {
+					if (device.deviceId) {
+						tempDevices.push(device);
+					}
+				});
+				micDevices = tempDevices;
+			}
+		);
+	} catch (err) {
+		if (err instanceof DOMException && err.name == "NotAllowedError") {
+			dispatch("error", i18n("audio.allow_recording_access"));
+		}
+		throw err;
+	}
+</script>
+
+<select
+	class="mic-select"
+	aria-label="Select input device"
+	disabled={micDevices.length === 0}
+>
+	{#if micDevices.length === 0}
+		<option value="">{i18n("audio.no_microphone")}</option>
+	{:else}
+		{#each micDevices as micDevice}
+			<option value={micDevice.deviceId}>{micDevice.label}</option>
+		{/each}
+	{/if}
+</select>
+
+<style>
+	.mic-select {
+		height: var(--size-8);
+		background: var(--block-background-fill);
+		padding: 0px var(--spacing-xxl);
+		border-radius: var(--radius-full);
+		font-size: var(--text-md);
+		border: 1px solid var(--neutral-400);
```

Comment:
nit: When highlighted the bottom of the border is a dark blue?

<img width="618" alt="image" src="https://github.com/gradio-app/gradio/assets/41651716/30c00bb5-b660-4d98-be5a-469846919c47">

Unrelated to this PR I think so we can do in a separate PR